src/Valit/Rules/ValitRule.cs

```diff
@@ -67,15 +69,16 @@ public IValitResult Validate(TObject @object)
         {
             @object.ThrowIfNull();
 
-            var property = _propertySelector(@object);
+            var property = _propertySelector.Compile().Invoke(@object);
             var hasAllConditionsFulfilled = true;
 
             foreach(var condition in _conditions)
                 hasAllConditionsFulfilled &= condition(@object);
 
             var isSatisfied = _predicate?.Invoke(property) != false;
+            var errors = _errors.Where(e => !e.IsDefault).Any() ? _errors.Where(e => !e.IsDefault) : _errors;
```

Comment:
Could be rewritten as `_errors.Any(e => !e.IsDefault)`, which is cleaner imo.

Reply:
Damn :D I always forget about it, sorry :D

Reply:
What do you mean by having multiple default messages? The fact it's stored inside ``IEnumerable<>`` ?

Reply:
If you do `rule.WithMessage(xxx, true);` multiple times the `ValitRule` would contain multiple default messages in `List<ValitRuleError> _errors`.

Reply:
Yes, but this overload is internal ;) User can only call the public one which calls the internal one with ``false`` beaneath.

Reply:
True, I was thinking the same - but it technically can be done.
Was just asking if that's OK.
I guess it it is 😄

Reply:
I hope so :D